controllers/cards.js

```diff
@@ -20,26 +19,44 @@ module.exports.postCard = (req, res) => { // добавление карточк
 };
 
 module.exports.deleteCardById = (req, res) => { // удаление карточки
-  Card.findByIdAndRemove(req.params.id)
-    .then((user) => res.send({ data: user }))
-    .catch((err) => res.status(500).send({ message: err.message }));
-  if (Card.findByIdAndRemove(req.params.id)) {
-    return res.status(404).send({ message: 'Нет пользователя с таким id' });
-  }
+  Card.findById(req.params.id)
+    .orFail(() => new NotFoundError('Карточка не найдена'))
+    .then((card) => {
+      if (!card.owner.equals(req.user._id)) {
+        return res.status(403).send({ message: 'Карточка добавлена не вами - удаление невозможно' });
```

Comment:
Можно лучше: создать отдельный класс ошибки для 403.
По аналогии с 401 и 404 можно создать отдельный класс и бросать исключение. Далее всю необходимую обработку сделает `.catch()` блок:
```suggestion
        return Promise.reject(new ForbiddenError('Карточка добавлена не вами - удаление невозможно'));
```
либо `throw`:
```suggestion
        throw new ForbiddenError('Карточка добавлена не вами - удаление невозможно');
```